Formula/b/backtrace_crashpad.rb

```diff
@@ -0,0 +1,77 @@
+    include.install "client/length_delimited_ring_buffer.h" => "backtrace/crashpad/client/length_delimited_ring_buffer.h"
+    include.install "client/crashpad_info.h" => "backtrace/crashpad/client/crashpad_info.h"
+    include.install "client/prune_crash_reports.h" => "backtrace/crashpad/client/prune_crash_reports.h"
+    include.install "client/simulate_crash_mac.h" => "backtrace/crashpad/client/simulate_crash_mac.h"
+    include.install "client/crash_report_database.h" => "backtrace/crashpad/client/crash_report_database.h"
+    include.install "client/client_argv_handling.h" => "backtrace/crashpad/client/client_argv_handling.h"
+    include.install "client/annotation.h" => "backtrace/crashpad/client/annotation.h"
+
+    mkdir "#{include}/backtrace/util/file"
+    include.install "util/file/file_io.h" => "backtrace/util/file/file_io.h"
+    mkdir "#{include}/backtrace/util/misc"
+    include.install "util/misc/capture_context.h" => "backtrace/util/misc/capture_context.h"
+    mkdir "#{include}/backtrace/base"
+    include.install "third_party/mini_chromium/mini_chromium/base/scoped_generic.h" => "backtrace/base/scoped_generic.h"
+    mkdir "#{include}/backtrace/base/mac"
+    include.install "third_party/mini_chromium/mini_chromium/base/mac/scoped_mach_port.h" => "backtrace/base/mac/scoped_mach_port.h"
+    mkdir "#{include}/backtrace/base/files"
+    include.install "third_party/mini_chromium/mini_chromium/base/files/file_path.h" => "backtrace/base/files/file_path.h"
+    include.install "third_party/mini_chromium/mini_chromium/base/files/scoped_file.h" => "backtrace/base/files/scoped_file.h"
+    mkdir "#{include}/backtrace/build"
+    include.install "third_party/mini_chromium/mini_chromium/build/build_config.h" => "backtrace/build/build_config.h"
+    include.install "third_party/mini_chromium/mini_chromium/build/buildflag.h" => "backtrace/build/buildflag.h"
+
+    cmake_config = <<-EOF
+add_library(backtrace::crashpad STATIC IMPORTED)
+set_target_properties(backtrace::crashpad PROPERTIES
+  IMPORTED_LOCATION "${CMAKE_CURRENT_LIST_DIR}/../../libbacktrace_crashpad_client.a")
+target_include_directories(backtrace::crashpad 
+  INTERFACE "${CMAKE_CURRENT_LIST_DIR}/../../../include/backtrace")
+set(BacktraceCrashpad_FOUND TRUE)
+    EOF
+    cmake_version_config = <<-EOF
+set(PACKAGE_VERSION "#{version}")
+if(PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION)
+  set(PACKAGE_VERSION_EXACT TRUE)
+endif()
+if(NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+  set(PACKAGE_VERSION_COMPATIBLE TRUE)
+else(NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+  set(PACKAGE_VERSION_UNSUITABLE TRUE)
+endif()
+    EOF
+    mkdir "#{lib}/cmake/BacktraceCrashpad"
+    File.write("#{lib}/cmake/BacktraceCrashpad/BacktraceCrashpadConfig.cmake", cmake_config)
+    File.write("#{lib}/cmake/BacktraceCrashpad/BacktraceCrashpadConfigVersion.cmake", cmake_version_config)
+  end
+
+  test do
+    system "#{bin}/backtrace_crashpad_handler", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@shraddhaarora Let's do the first option: 

> If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.